en/docs/develop-components/develop-services/develop-a-grpc-service.md

```diff
@@ -0,0 +1,210 @@
+3. Go to the **Service** card and click **Create**.
+4. You can enter a unique name and a description of the service. For this guide, let's enter the following values:
+
+    | **Field**       | **Value**               |
+    |-----------------|-------------------------|
+    | **Name**        | `Go gRPC Server`        |
+    | **Description** | `Sends greetings`       |
+
+5. Click **Next**.
+6. To allow Choreo to connect to your GitHub account, click **Authorize with GitHub**.
+7. If you have not already connected your GitHub repository to Choreo, enter your GitHub credentials, and select the repository you created by forking [https://github.com/wso2/choreo-examples](https://github.com/wso2/choreo-examples) to install the [Choreo GitHub App](https://github.com/marketplace/choreo-apps).
+
+    !!! info
+         The **Choreo GitHub App** requires the following permissions:
+
+         - Read and write access to code and pull requests.
+         - Read access to issues and metadata.
+             
+          You can [revoke access](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/reviewing-your-authorized-integrations#reviewing-your-authorized-github-apps) if you do not want Choreo to have access to your GitHub account. However, write access is only used to send pull requests to a user repository. Choreo will not directly push any changes to a repository.
+
+8. In the **Connect Repository** pane, enter the following information:
+
+    | **Field**               | **Description**                 |
+    |-------------------------|---------------------------------|
+    | **GitHub Account**      | Your account                    |
+    | **GitHub Repository**   | **`choreo-sample-apps`**        |
+    | **Branch**              | **`main`**                      |
+    | **Build Preset**        | **Dockerfile**                  |
+    | **Dockerfile path**     | **go/grpc/Dockerfile.server** |
+    | **Docker context path** | **go/grpc**                     |
+
+
+    !!! info
+        1.  To successfully build your container with Choreo, it is essential to explicitly define a User ID (UID) under the USER instruction in your Dockerfile. For reference, see [sample Dockerfile](https://github.com/wso2/choreo-sample-apps/blob/main/go/greeter/Dockerfile).
+        To ensure that the defined USER instruction is valid, it must conform to the following conditions:
+            - A valid User ID is a numeric value between 10000-20000, such as `10001` or `10500`.
+            - Usernames are considered invalid and should not be used. For example, `my-custom-user-12221` or `my-custom-user` are invalid User IDs.
+
+        2. The Dockerfile utilized in this guide is a Multi-stage Dockerfile, which is designed to keep the final image size small and provides the ability to build the application with a specific version of tools and libraries.
```

Comment:
It is not necessary to capitalize M in "multi-stage" right?

Reply:
true. Removed.